src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -49,9 +54,10 @@ export const buildSafe = async (safeAddress: string): Promise<SafeRecordProps> =
  *
  * @param {string} safeAddress
  */
+let isFirstSafeFetch = true
```

Comment:
Will this still work when you switch safes?
I would expect this "do once per mount" logic to be implemented via an effect hook with [] dependencies.